filebeat/input/filestream/input_integration_test.go

```diff
@@ -67,6 +70,50 @@ func TestFilestreamCloseRenamed(t *testing.T) {
 	env.requireOffsetInRegistry(testlogName, len(newerTestlines))
 }
 
+// test_close_removed from test_harvester.py
+func TestFilestreamCloseRemoved(t *testing.T) {
```

Comment:
In later PRs I add a function named `waitUntilHarvesterDone` which checks indirectly if the `Harvester` has finished. It is indirect as in I check if the client has been closed because the `defaultHarvesterGroup` closes the client of a `Harvester` when the `Harvester` returns. See in the original PR: https://github.com/elastic/beats/pull/24397/files#diff-b8e1598192adc8a2bbb9fb270c9115c6527b91ce8b39b16ee08b43d56e4eebc7R240

Would that check enough here?

Reply:
Yes, this check would help. No need to write to a new file.